server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorFactory.java

```diff
@@ -290,6 +290,10 @@ protected Aggregator doCreateInternal(Aggregator parent, CardinalityUpperBound c
             // counting
             bucketCountThresholds.setShardSize(BucketUtils.suggestShardSideQueueSize(bucketCountThresholds.getRequiredSize()));
         }
+        // We should scale the shard min doc count to account for the reduction due to sampling
+        getSamplingContext().ifPresent(
+            samplingContext -> bucketCountThresholds.setShardMinDocCount(samplingContext.scale(bucketCountThresholds.getShardMinDocCount()))
```

Comment:
I'd like to have some tests around this.  Terms does non obvious things with `ShardMinDocCount` (see https://github.com/elastic/elasticsearch/issues/70664 for example) and I want to make sure this honors that behavior.